src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.cpp

```diff
@@ -398,109 +398,124 @@ void SinglePhaseHybridFVM::saveAquiferConvergedState( real64 const & time,
                                                     arrayView1d< real64 const > const & localRhs )
 {
-  // here we compute the cell-centered residual norm in the derived class
-  // to avoid duplicating a synchronization point
+  GEOSX_MARK_FUNCTION;
 
-  // get a view into local residual vector
+  real64 localResidualNorm = 0.0;
+  real64 localResidualNormalizer = 0.0;
 
+  globalIndex const rankOffset = dofManager.rankOffset();
   string const elemDofKey = dofManager.getKey( viewKeyStruct::elemDofFieldString() );
   string const faceDofKey = dofManager.getKey( extrinsicMeshData::flow::facePressure::key() );
 
-  globalIndex const rankOffset = dofManager.rankOffset();
-
-  // local residual
-  real64 localResidualNorm[4] = { 0.0, 0.0, 0.0, 0.0 };
-  real64 globalResidualNorm[4] = { 0.0, 0.0, 0.0, 0.0 };
-
-  // 1. Compute the residual for the mass conservation equations
-
-  // compute the norm of local residual scaled by cell pore volume
-
-  real64 defaultViscosity = 0; // for the normalization of the face residuals
-  localIndex subRegionCounter = 0;
-
   forMeshTargets( domain.getMeshBodies(), [&] ( string const &,
                                                 MeshLevel const & mesh,
                                                 arrayView1d< string const > const & regionNames )
   {
-    StencilAccessors< extrinsicMeshData::elementVolume > flowAccessors( mesh.getElemManager(), getName() );
+
+    ElementRegionManager const & elemManager = mesh.getElemManager();
     FaceManager const & faceManager = mesh.getFaceManager();
+    real64 defaultViscosity = 0;
+    integer subRegionCounter = 0;
 
-    mesh.getElemManager().forElementSubRegions< ElementSubRegionBase >( regionNames, [&]( localIndex const,
-                                                                                          ElementSubRegionBase const & subRegion )
+    // step 1: compute the residual for the element-based mass conservation equations
+
+    elemManager.forElementSubRegions< ElementSubRegionBase >( regionNames,
+                                                              [&]( localIndex const,
+                                                                   ElementSubRegionBase const & subRegion )
     {
+      real64 subRegionResidualNorm[1]{};
+      real64 subRegionResidualNormalizer[1]{};
 
-      arrayView1d< globalIndex const > const & elemDofNumber = subRegion.template getReference< array1d< globalIndex > >( elemDofKey );
-      arrayView1d< integer const > const & elemGhostRank = subRegion.ghostRank();
-      arrayView1d< real64 const > const & volume = subRegion.getElementVolume();
+      string const & fluidName = subRegion.getReference< string >( viewKeyStruct::fluidNamesString() );
+      SingleFluidBase const & fluid = getConstitutiveModel< SingleFluidBase >( subRegion, fluidName );
+      defaultViscosity += fluid.defaultViscosity();
+      subRegionCounter++;
```

Comment:
It was already there, but I do not quite understand why we divide by the number of sub regions.

Reply:
Yes, here I was trying to get a default value of viscosity for normalization purposes. But since the default viscosity lives in the constitutive model (attached to the subregion), I averaged by the number of subregions to get a representative value (instead of picking the default viscosity of a particular subregion).